From f4a11c782086ea852d338ec83adb1254175e8873 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maxime=20M=C3=A9r=C3=A9?= <maxime.mere@foss.st.com>
Date: Thu, 20 Feb 2025 09:57:47 +0100
Subject: [PATCH 797/841] fix(stm32mp2-fdts): put DDR firmware under flag in
 stm32mp2-cot-descriptors.dtsi
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

When DDR firmware is not present in FIP it must not be present in the
chain of trust descriptor. Put the ddr_fw node under
STM32MP_DDR_FIP_IO_STORAGE to fix that.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 63df44afb38c14499fe6a83f5076dae9ab9c0123 fix(stm32mp2-fdts): put DDR firmware under flag in stm32mp2-cot-descriptors.dtsi]
Change-Id: I264e328bba9c10c7bad7ca64a72883bf05779460
Signed-off-by: Maxime Méré <maxime.mere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/438658
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 fdts/stm32mp2-cot-descriptors.dtsi | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/fdts/stm32mp2-cot-descriptors.dtsi b/fdts/stm32mp2-cot-descriptors.dtsi
index dd74b164a..a30855a8e 100644
--- a/fdts/stm32mp2-cot-descriptors.dtsi
+++ b/fdts/stm32mp2-cot-descriptors.dtsi
@@ -170,11 +170,13 @@ cot {
 			hash = <&nt_world_bl_hash>;
 		};
 
+#if STM32MP_DDR_FIP_IO_STORAGE
 		ddr_fw {
 			image-id = <DDR_FW_ID>;
 			parent = <&stm32mp_cfg_cert>;
 			hash = <&ddr_fw_hash>;
 		};
+#endif
 	};
 };
 
-- 
2.47.2

